src/utils/file_upload.rs

```diff
@@ -436,7 +437,10 @@ fn build_artifact_bundle(
         bundle.set_attribute("dist".to_owned(), dist.to_owned());
     }
 
-    for file in files.values() {
+    let mut files_sorted = files.values().collect::<Vec<_>>();
+    files_sorted.sort_by_key(|file| file.url.clone());
```

Comment:
Is there a reason you need to clone?

The docs say:

> For expensive key functions (e.g. functions that are not simple property accesses or basic operations), [sort_by_cached_key](https://doc.rust-lang.org/std/primitive.slice.html#method.sort_by_cached_key) is likely to be significantly faster, as it does not recompute element keys.

Reply:
I had some lifetime issue and fixed it by cloning. But borrowing actually works just fine.